File: be/pnt_spring/src/main/java/com/pnt/pnt_spring/domain/auth/api/req/SignupRequest.java
package com.pnt.pnt_spring.domain.auth.api.req;

import jakarta.validation.constraints.NotBlank;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Getter
@NoArgsConstructor
public class SignupRequest {
    @NotBlank(message = "아이디는 필수입니다.")
    private String id; // user ID 아니고 로그인 ID

    @NotBlank(message = "비밀번호는 필수입니다.")
    private String password;

    @NotBlank(message = "비밀번호 확인은 필수입니다.")
    private String passwordConfirm;

    @NotBlank(message = "닉네임은 필수입니다.")
    private String nickname;

    private String email;
    private LocalDate birth;
    private String avatarUrl;

    public String getAvatarUrl() {
        if (avatarUrl == null) {
            return "default.png";
        }
        return avatarUrl;
    }
}

File: fe/D104/app/src/main/java/com/d104/pnt/data/remote/model/request/SignupRequest.kt
package com.d104.pnt.data.remote.model.request

import com.google.gson.annotations.SerializedName

/**
 * 회원가입 요청
 */
data class SignupRequest(
    @SerializedName("id")
    val id: String,

    @SerializedName("password")
    val password: String,

    @SerializedName("passwordConfirm")
    val passwordConfirm: String,

    @SerializedName("nickname")
    val nickname: String,

    @SerializedName("email")
    val email: String,

    @SerializedName("birth")
    val birth: String,

    @SerializedName("avatarUrl")
    val avatarUrl: String? = null
)


/**
 * 아이디 중복 체크 요청
 */
data class CheckDuplicateRequest(
    @SerializedName("id")
    val id: String,
)